From 25f4e7ea94355f83b087f937a2aec7206cce562f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:07 -0400
Subject: [usb] hcd: remove unnecessary local_irq_save

Message-id: <1410887463-20674-56-git-send-email-dzickus@redhat.com>
Patchwork-id: 95145
O-Subject: [RHEL7 PATCH 055/591] usb/hcd: remove unnecessary local_irq_save
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 88ed9fd50e573fee332ca1e07641102b59f743fe
Author: Michael Opdenacker <michael.opdenacker@free-electrons.com>
Date:   Sun Oct 13 09:14:19 2013 +0200

    usb/hcd: remove unnecessary local_irq_save

    Remove the use of local_irq_save() and IRQF_DISABLED, no longer needed since
    interrupt handlers are always run with interrupts disabled on the
    current CPU.

    Tested successfully with 3.12.0-rc4 on my PC. Didn't find
    any issue because of this change.

    Signed-off-by: Michael Opdenacker <michael.opdenacker@free-electrons.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hcd.c b/drivers/usb/core/hcd.c
index 6ec8dda..6bffb8c 100644
--- a/drivers/usb/core/hcd.c
+++ b/drivers/usb/core/hcd.c
@@ -2331,15 +2331,8 @@ EXPORT_SYMBOL_GPL(usb_bus_start_enum);
 irqreturn_t usb_hcd_irq (int irq, void *__hcd)
 {
  struct usb_hcd  *hcd = __hcd;
- unsigned long  flags;
  irqreturn_t  rc;
 
- /* IRQF_DISABLED doesn't work correctly with shared IRQs
-  * when the first handler doesn't use it.  So let's just
-  * assume it's never used.
-  */
- local_irq_save(flags);
-
  if (unlikely(HCD_DEAD(hcd) || !HCD_HW_ACCESSIBLE(hcd)))
   rc = IRQ_NONE;
  else if (hcd->driver->irq(hcd) == IRQ_NONE)
@@ -2347,7 +2340,6 @@ irqreturn_t usb_hcd_irq (int irq, void *__hcd)
  else
   rc = IRQ_HANDLED;
 
- local_irq_restore(flags);
  return rc;
 }
 EXPORT_SYMBOL_GPL(usb_hcd_irq);
@@ -2554,13 +2546,6 @@ static int usb_hcd_request_irqs(struct usb_hcd *hcd,
 
  if (hcd->driver->irq) {
 
-  /* IRQF_DISABLED doesn't work as advertised when used together
-   * with IRQF_SHARED. As usb_hcd_irq() will always disable
-   * interrupts we can remove it here.
-   */
-  if (irqflags & IRQF_SHARED)
-   irqflags &= ~IRQF_DISABLED;
-
   snprintf(hcd->irq_descr, sizeof(hcd->irq_descr), "%s:usb%d",
     hcd->driver->description, hcd->self.busnum);
   retval = request_irq(irqnum, &usb_hcd_irq, irqflags,
-- 
1.7.1